feat(Dashboard): Add a dashboard for failed jobs

rnjane opened the pull request:
#### What does this PR do?
Creates a dashboard for all failed jobs.
#### Description of Task to be completed.
GIVEN that a user has been notified of a failed job
WHEN the user logs into the dashboard
THEN they should have a view listing all their failed jobs which they have been notified about.
#### What are the relevant pivotal tracker stories?
User requires a new dashboard tab for unresolved stuff - #150627363
#### Screenshot
![image](https://user-images.githubusercontent.com/22521168/30210697-34332b9e-94a6-11e7-96de-4f36e69e90d0.png)


[Delivers #150627363]
@emugaya @margierain

Commit:
feat(Dashboard): Add a dashboard for failed jobs

- Add a link to the main dashboard that takes a user to a dashboard with all the failed jobs(only).

[Delivers #150627363]

aenkya reviewed (commented):
Did you write any tests for this?

emugaya commented:
Please add tests for this feature

Commit:
test(checks): add tests for failed jobs dashboard

- add test for the newly created failed checks dashboard

[Delivers #150627363]

Commit:
test(checks): tests for failed checks dashboard

- assert failed checks dashboard only has failed checks

[Delivers #150627363]

NgangaMaryanne commented on hc/front/tests/test_my_checks.py:
```diff
@@ -58,3 +58,19 @@ def test_it_shows_amber_check(self):
         # Mobile
         self.assertContains(r, "label-warning")
+
+    def test_failed_checks_view(self):
```
This is good but since this is a failed checks view, shouldnt you assert that failed checks actually exist, ie `contains "icon-down"`?

NgangaMaryanne commented on hc/front/tests/test_my_checks.py:
```diff
@@ -58,3 +58,19 @@ def test_it_shows_amber_check(self):
+        self.check.save()
+
+        self.client.login(username="alice@example.org", password="password")
+        r = self.client.get("/checks/failed/")
```
Please use descriptive variable names (I can not be sure what r means).

NgangaMaryanne commented on hc/front/views.py:
```diff
@@ -60,6 +60,43 @@ def my_checks(request):
 
+@login_required
+def my_failed_checks(request):
+    q = Check.objects.filter(user=request.team.user).order_by("created")
```
descriptive variable names are highly advised. Please change this.

NgangaMaryanne commented on templates/front/my_failed_checks.html:
```diff
@@ -0,0 +1,284 @@
+<script src="{% static 'js/clipboard.min.js' %}"></script>
+<script src="{% static 'js/checks.js' %}"></script>
+{% endcompress %}
+{% endblock %}
```
Generally, good job done on this feature, logic, tests and template.

Commit:
feat(checks):Failed jobs dashboard

- Rename variable names, to be more descriptive.
- Improve test logic.

[Delivers #150627363]

NgangaMaryanne commented:
LGTM
